pkg/k8s/watchers/pod.go

```diff
@@ -314,32 +314,32 @@ func (k *K8sWatcher) deleteK8sPodV1(pod *slim_corev1.Pod) error {
 	return err
 }
 
-func genServiceMappings(pod *slim_corev1.Pod, podIPs []string) []loadbalancer.SVC {
+func (k *K8sWatcher) genServiceMappings(pod *slim_corev1.Pod, podIPs []string, logger *logrus.Entry) []loadbalancer.SVC {
 	var svcs []loadbalancer.SVC
 	for _, c := range pod.Spec.Containers {
 		for _, p := range c.Ports {
-			if p.HostPort == 0 {
+			var fes4 []loadbalancer.L3n4AddrID
+			var fes6 []loadbalancer.L3n4AddrID
+
+			if p.HostPort <= 0 {
 				continue
 			}
-			feIP := net.ParseIP(p.HostIP)
-			if feIP == nil {
-				feIP = net.ParseIP(pod.Status.HostIP)
+
+			if int(p.HostPort) >= option.Config.NodePortMin &&
+				int(p.HostPort) <= option.Config.NodePortMax {
+				logger.Warningf("The requested hostPort %d is colliding with the configured NodePort range [%d, %d]. Ignoring.",
+					p.HostPort, option.Config.NodePortMin, option.Config.NodePortMax)
+				continue
 			}
+
 			proto, err := loadbalancer.NewL4Type(string(p.Protocol))
 			if err != nil {
 				continue
 			}
-			fe := loadbalancer.L3n4AddrID{
-				L3n4Addr: loadbalancer.L3n4Addr{
-					IP: feIP,
-					L4Addr: loadbalancer.L4Addr{
-						Protocol: proto,
-						Port:     uint16(p.HostPort),
-					},
-				},
-				ID: loadbalancer.ID(0),
-			}
-			bes := make([]loadbalancer.Backend, 0, len(podIPs))
+
+			bes4 := make([]loadbalancer.Backend, 0, len(podIPs))
+			bes6 := make([]loadbalancer.Backend, 0, len(podIPs))
```

Comment:
We are unnecessarily allocating more memory than we really need here. `bes6` won't have `len(podIPs)` if all backends are a v4 for example.